rustls/src/client/handy.rs

```diff
@@ -80,27 +80,36 @@ mod cache {
     impl ClientSessionMemoryCache {
         /// Make a new ClientSessionMemoryCache.  `size` is the
         /// maximum number of stored sessions.
+        #[cfg(feature = "std")]
         pub fn new(size: usize) -> Self {
             let max_servers = size.saturating_add(MAX_TLS13_TICKETS_PER_SERVER - 1)
                 / MAX_TLS13_TICKETS_PER_SERVER;
             Self {
                 servers: Mutex::new(limited_cache::LimitedCache::new(max_servers)),
             }
         }
+
+        /// Make a new ClientSessionMemoryCache.  `size` is the
+        /// maximum number of stored sessions.
+        #[cfg(not(feature = "std"))]
+        pub fn new<M: crate::lock::MakeMutex>(size: usize) -> Self {
+            let max_servers = size.saturating_add(MAX_TLS13_TICKETS_PER_SERVER - 1)
+                / MAX_TLS13_TICKETS_PER_SERVER;
+            Self {
+                servers: Mutex::new::<M>(limited_cache::LimitedCache::new(max_servers)),
+            }
+        }
     }
 
     impl super::client::ClientSessionStore for ClientSessionMemoryCache {
         fn set_kx_hint(&self, server_name: ServerName<'static>, group: NamedGroup) {
-            self.servers
-                .lock()
-                .unwrap()
-                .get_or_insert_default_and_edit(server_name, |data| data.kx_hint = Some(group));
+            let mut servers = self.servers.lock().unwrap();
```

Comment:
I think these changes can be reverted? (Or split off into a separate commit -- it's just a style thing?)

Reply:
Agreed. I will tidy up

Reply:
Cleaned up!